scalajs/src/main/scala/clue/js/WebSocketJSBackend.scala

```diff
@@ -16,77 +16,78 @@ import org.scalajs.dom.raw.Event
 import org.scalajs.dom.raw.MessageEvent
 import org.scalajs.dom.raw.WebSocket
 import sttp.model.Uri
-import cats.effect.concurrent.Ref
+import cats.effect.Ref
+import cats.effect.std.Dispatcher
 
 /**
  * Streaming backend for JS WebSocket.
  */
-final class WebSocketJSBackend[F[_]: Effect: Logger] extends WebSocketBackend[F] {
+final class WebSocketJSBackend[F[_]: Async: Logger] extends WebSocketBackend[F] {
   private val Protocol = "graphql-ws"
 
   override def connect(
     uri:          Uri,
     handler:      PersistentBackendHandler[F, WebSocketCloseEvent],
-    connectionId: ConnectionId
+    connectionId: ConnectionId,
+    dispatcher:   Dispatcher[F]
   ): F[PersistentConnection[F, WebSocketCloseParams]] =
     for {
       isOpen     <- Ref[F].of(false)
       isErrored  <- Ref[F].of(false)
       connection <-
-        Async[F].async[PersistentConnection[F, WebSocketCloseParams]] { cb =>
+        Async[F].async_[PersistentConnection[F, WebSocketCloseParams]] { cb =>
           val ws = new WebSocket(uri.toString, Protocol)
 
           ws.onopen = { _: Event =>
-            (
+            val open: F[Unit] = (
               for {
                 _ <- isOpen.set(true)
                 _ <- s"WebSocket open for URI [$uri]".traceF
               } yield cb(new WebSocketJSConnection(ws).asRight)
             ).uncancelable
-              .runAsync(_ => IO.unit)
-              .unsafeRunSync()
+            dispatcher.unsafeRunAndForget(open)
```

Comment:
Using `dispatcher` is the recommended alternative to run effects when interacting with a foreign interface.
Still it would be good to test it for real but I guess that needs to wait for explore to be ported

Reply:
I see. Maybe we can have the `Dispatcher.use` call here instead of passing it from the caller. It seems this would simplify the `ApolloClient` logic and we can remove the `dispatcher` parameter.

Quoting the docs:
`An instance of Dispatcher is very cheap - it allocates a single fiber so it is encouraged to instantiate it where necessary rather than wiring a single instance throughout an application.`

Reply:
Right it is really up to use whether we create locally or pass it along. In this case I thought we could pass it at this level and create it at `ApolloClient`. I'm not sure of the pros/cons

Reply:
If we use `Resource.use` that will produce a `F[A]` rather than `Unit` and that has to be executed by the `onopen` method. I think we have to use the `unsafe` methods here

Reply:
I agree that we have to call `unsafe` here.

Can we create the `Dispatcher` instance here? I see as a pro that the `WebSoketBackend` API, as well as the `ApolloClient`, will both be kept pure, without the need of them to know about the `Dispatcher` at all.

Reply:
Ok, let's try that.

Reply:
Ok done